io/eolearn/io/extra/meteoblue.py

```diff
@@ -76,14 +76,19 @@ def __init__(
         self.time_difference = time_difference
 
     @staticmethod
-    def _prepare_bbox(eopatch, bbox):
-        """Prepares a bbox from input parameters"""
-        if not eopatch.bbox and not bbox:
+    def _get_modified_eopatch(eopatch: Optional[EOPatch], bbox: Optional[BBox]) -> EOPatch:
+        if bbox is not None:
+            if eopatch is None:
+                eopatch = EOPatch(bbox=bbox)
+            elif eopatch.bbox is None:
+                eopatch.bbox = bbox
+            elif eopatch.bbox != bbox:
+                raise ValueError("Provided eopatch.bbox and bbox are not the same")
+            return eopatch
+
+        if eopatch is None or eopatch.bbox is None:
             raise ValueError("Bounding box is not provided")
-        if eopatch.bbox and bbox and eopatch.bbox != bbox:
-            raise ValueError("Provided eopatch.bbox and bbox are not the same")
-
-        return bbox or eopatch.bbox
+        return eopatch
```

Comment:
As mentioned, this functionality seems almost identical to the one in https://github.com/sentinel-hub/eo-learn/pull/565#discussion_r1109858198, perhaps we could do something about it, like extract the logic to a util or something along those lines?

Reply:
According to the other replies I guess it makes sense to leave this for now as is, and we can later try to find some more elegant ways when we see which parts drop out.

Reply:
yeah, possibly.